primeqa/ir/dense/colbert_top/colbert/modeling/factory.py

```diff
@@ -14,106 +16,92 @@
-def get_query_tokenizer(model_type, maxlen, attend_to_mask_tokens):
-    model_dir = None
-
-    # if it is a directory, load json file to get the model type
-    if os.path.isdir(model_type):
-        model_dir = model_type
-        json_file = model_type + '/config.json'
-        print_message(f"json file (get_query_tokenizer): {json_file}")
-        with open(json_file) as file:
-            data = json.load(file)
-        model_type = data["_name_or_path"]
-
-    print_message(f"get query model type: {model_type}")
-
-    if model_type=='bert-base-uncased' or model_type=='bert-large-uncased':
-        return QueryTokenizer(maxlen,model_type, attend_to_mask_tokens)
-    elif model_type=='tinybert':
-        return QueryTokenizer(maxlen, 'bert-base-uncased',attend_to_mask_tokens)
-    elif model_type=='xlm-roberta-base' or model_type=='xlm-roberta-large':
-        return QueryTokenizerXLMR(maxlen, model_type)
-    elif model_type=='roberta-base' or model_type=='roberta-large':
-        return QueryTokenizerRoberta(maxlen, model_type)
+def get_query_tokenizer(name, colbert_config):
+    model_type = colbert_config.model_type
+    maxlen = colbert_config.query_maxlen
+    attend_to_mask_tokens = colbert_config.attend_to_mask_tokens
+
+    print_message(f"factory model type: {model_type}")
+
+    if model_type == 'bert':
+        return QueryTokenizer(maxlen, name, attend_to_mask_tokens)
+    elif model_type == 'xlm-roberta':
+        return QueryTokenizerXLMR(maxlen, name)
+    elif model_type == 'roberta':
+        return QueryTokenizerRoberta(maxlen, name)
     else:
-        raise NotImplementedError
+        raise NotImplementedError(f"Model type: {model_type} is not supported.")
 
 #----------------------------------------------------------------
-def get_doc_tokenizer(model_type, maxlen):
-    model_dir = None
-
-    # if it is a directory, load json file to get the model type
-    if os.path.isdir(model_type):
-        model_dir = model_type
-        json_file = model_type + '/config.json'
-        print_message(f"json file (get_doc_tokenizer): {json_file}")
-        with open(json_file) as file:
-            data = json.load(file)
-        model_type = data["_name_or_path"]
-
-
-    print_message(f"get doc model type: {model_type}")
-
-    if model_type=='bert-base-uncased' or model_type=='bert-large-uncased':
-        return DocTokenizer(maxlen, model_type)
-    elif model_type=='tinybert':
-        return DocTokenizer(maxlen, 'bert-base-uncased')
-    elif model_type=='xlm-roberta-base' or model_type=='xlm-roberta-large':
-        return DocTokenizerXLMR(maxlen, model_type)
-    elif model_type=='roberta-base' or model_type=='roberta-large':
-        return DocTokenizerRoberta(maxlen, model_type)
+def get_doc_tokenizer(name, colbert_config, is_teacher=False):
+    model_type = colbert_config.model_type
+    maxlen = config.teacher_doc_maxlen if is_teacher else colbert_config.doc_maxlen
```

Comment:
Should this be `maxlen = colbert_config.teacher_doc_maxlen if is_teacher else colbert_config.doc_maxlen` ?

Reply:
Yes, I will fix this.